src/core/labeling/qgspallabeling.h

```diff
@@ -636,6 +636,9 @@ class CORE_EXPORT QgsPalLayerSettings
     //! True if label rotation should be preserved during label pin/unpin operations.
     bool preserveRotation = true;
 
+    //! Unit for rotation of labels.
+    QgsUnitTypes::AngleUnit rotationUnit = QgsUnitTypes::AngleDegrees;
```

Comment:
Can you make the member variable private and add getter/setter functions? Having public member variables makes things very tricky with the stable API restriction

Reply:
At line 18 it is stated that this file is not part of the stable API. Anyway I would have nothing against moving it to private but all other properties are in the public section too. For code consistency I would leave it there for now and add a TODO comment for QGIS 4 that all this members should become private. What do you think?

Reply:
That's an extremely old comment which should definitely be removed -- this class has been stable since early 2.x. I've been slowly cleaning up this class but it's taken a lot of messy code because of the public members, which is why I don't want any more added.